src/parse.c

```diff
@@ -3134,6 +3134,23 @@ node_is_nulled_out(yaml_document_t* doc, yaml_node_t* node, const char* key_pref
     return TRUE;
 }
 
+/*
+ * Remove a given netdef from the parser state
+ */
+STATIC void
+netplan_parser_drop_netdef(NetplanParser* npp, NetplanNetDefinition *netdef)
+{
+    if(npp->parsed_defs) {
+        g_hash_table_steal(npp->parsed_defs, netdef->id);
+    }
+    if(npp->ordered) {
+        npp->ordered = g_list_remove(npp->ordered, netdef);
+    }
+
+    reset_netdef(netdef, NETPLAN_DEF_TYPE_NONE, NETPLAN_BACKEND_NONE);
+    g_free(netdef);
```

Comment:
question: wouldn't that create dangling pointers?

IIRC we have cross-references on netdefs in cases of bridges and the likes, and we try to resolve them on the spot. If that is still true, we should at least add big fat warnings to the docs of this function so that our future selves don't start using it wrongly. Conversely, since we're only dropping the currently-parsed netdef for now we could actually make this function about that one special case?

Related question: don't we write backreferences *in* previously defined netdefs in some cases? It's been a long while since I've looked at the parser code, but I dimly remember that in some cases we go back and modify existing netdefs while parsing other ones, possibly to add a pointer to the new one.

Reply:
Yes, definitely. In the end I didn't do it. I decided to leave the netdef "half parsed" in the parser state in case of failure. It makes things simpler and increase the chances you'll have a working configuration in the end (I think).

If we end up generating invalid back end configuration due to this, both NM and networkd will reject the file so it should be fine.